### PR TITLE
General table

### PR DESCRIPTION
**Pull request by ArteGon**
I did the following:
- created table for our application
- created structure for  events
- refactoring general code

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
 import Staff from './components/staff/staff';
 import Authorization from './components/authorization/authorization';
 import Navbar from './components/navbar/navbar';
 import Error from './components/path_error/error';
+import Events from './components/Events/Events';
 
 const App: React.FC = () => {
     return (
-        <BrowserRouter>
+        <div>
```
Comment (DmitryPateew) [inline src/App.tsx:13]: Why do you use tag div instead of browser router?

Comment (ArteGon) [reply src/App.tsx:13]: When I have another structure, I created <Route>  component  into my <Event> component and I didn't want to duplicate <BrouserRouter> and brought up a level

**Review by DmitryPateew — DISMISSED** [review]

File: src/components/common/Table/ColumnHidePage/ColumnHidePage.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      >
+        <div className={classes.columns_pop_over}>
+          <Typography className={classes.popover_title}>Visible columns</Typography>
+          <div className={classes.grid}>
+            {hideableColumns.map((column) => (
+              <FormControlLabel
+                key={column.id}
+                control={
+                  <Checkbox
+                    value={`${column.id}`}
```
Comment (ZmicerBoksha) [inline src/components/common/Table/ColumnHidePage/ColumnHidePage.tsx]:
```suggestion
                    value={column.id}
```

File: src/App.tsx [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
             <Navbar />
             <Switch>
                 <Route exact path='/' component={Dashboard} />
-                <Route exact path='/events' component={Events} />
+                <Route   
+                    path='/events' 
+                    render={() => <Events />}
```
Comment (ZmicerBoksha) [inline src/App.tsx]:
```suggestion
                    component={Events}
```

**Comment by ZmicerBoksha** [comment]
Please use @types library for react-table instead of copy-paste typings to the current repo. (I mean src/components/common/react-table-config.d.ts file)
`npm install --save @types/react-table`

File: src/components/common/Table/ColumnHidePage/ColumnHidePage.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+const ColumnHidePage: FunctionComponent<ColumnHidePageProps> = ({ instance }) => {
+  const { allColumns, toggleHideColumn } = instance;
+
+  const classes = useStyle({});
+
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+  const [columnsOpen, setColumnsOpen] = useState(false);  
+
+  const handleColumnsClick = (event: React.MouseEvent<HTMLButtonElement>) => {
```
Comment (vchepurnaya) [inline src/components/common/Table/ColumnHidePage/ColumnHidePage.tsx:41]: add an import statement (import React from 'react') || (import { MouseEvent } from 'react' and delete word React)

File: src/components/common/Table/ColumnHidePage/ColumnHidePage.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+})
+
+type ColumnHidePageProps = {
+  instance: TableInstance
+}
+
+const ColumnHidePage: FunctionComponent<ColumnHidePageProps> = ({ instance }) => {
+  const { allColumns, toggleHideColumn } = instance;
+
+  const classes = useStyle({});
```
Comment (vchepurnaya) [inline src/components/common/Table/ColumnHidePage/ColumnHidePage.tsx]: you aren't use props, so you can delete {}

File: src/components/common/Table/TableFilters/TableFilters.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+const TableFilters: FunctionComponent<TableFiltersProps> = ({ instance }) => {
+  const { allColumns, setAllFilters } = instance;
+
+  const classes = useStyles({});
+
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+  const [filtersOpen, setFiltersOpen] = useState(false);
+
+  const handleFiltersClick = (event: React.MouseEvent<HTMLButtonElement>) => {
```
Comment (vchepurnaya) [inline src/components/common/Table/TableFilters/TableFilters.tsx]: add an import statement (import React from 'react') || (import { MouseEvent } from 'react' and delete word React)

File: src/components/common/Table/TableToolbar/TableToolbar.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  return (
+    <Toolbar className={classes.toolbar}>
+      <div className={classes.left_icons}>
+        {onAdd && 
+          <Tooltip title="Add new event">
+            <div>
+              <IconButton 
+                onClick={() => onAdd(instance)} 
+                aria-label="Add new event"
+                disabled={!(Object.keys(state.selectedRowIds).length === 0)}
```
Comment (vchepurnaya) [inline src/components/common/Table/TableToolbar/TableToolbar.tsx]:
```suggestion
                disabled={Object.keys(state.selectedRowIds).length !== 0}
```

**Review by vchepurnaya — COMMENTED** [review]
Check your files. In some files you need to add an import statement (import React from 'react') || (import { MouseEvent } from 'react' and delete word React)

**Review by ilonasadouskaya — DISMISSED** [review]

File: src/components/common/Table/TableToolbar/TableToolbar.tsx (hunk 7 above)
Comment (ZmicerBoksha) [reply src/components/common/Table/TableToolbar/TableToolbar.tsx]: May be will be enough
`Object.keys(state.selectedRowIds).length`

**Review by ZmicerBoksha — APPROVED** [review]

**Review by vchepurnaya — APPROVED** [review]

**Review by DmitryPateew — APPROVED** [review]

**Pull request merged**